paasta_tools/api/views/autoscaler.py

```diff
@@ -45,6 +46,25 @@ def update_autoscaler_count(request):
     service = request.swagger_data.get('service')
     instance = request.swagger_data.get('instance')
     desired_instances = request.swagger_data.get('json_body')['desired_instances']
+
+    try:
+        service_config = load_marathon_service_config(
+            service=service,
+            instance=instance,
+            cluster=settings.cluster,
+            soa_dir=settings.soa_dir,
+            load_deployments=False,
+        )
+    except:
+        error_message = 'Unable to load service config for %s.%s' % (service, instance)
+        raise ApiFailure(error_message, 404)
+
+    max_instances = service_config.get_max_instances()
+    min_instances = service_config.get_min_instances()
+    if desired_instances > max_instances or desired_instances < min_instances:
+        error_message = 'Incorrect desired_instances %d: min_instances %d max_instances %d' % (desired_instances, min_instances, max_instances)
+        raise ApiFailure(error_message, 400)
```

Comment:
I don't really like this because this requires anyone who implements the autoscaler to be aware of the limits in yelpsoa-configs as well as handle 400 errors. I like the idea of returning `instance_count=service_config.get_instances()`, which is limited to the service's min- and max-instances. That way if an autoscaler postal an out-of-bounds instance count, they still get an HTTP 200 that tells them authoritatively what the current desired instance count . That way if an autoscaler is going 'oh my god we need 500000 instances' PaaSTA will still scale up to max_instances and report back to the autoscaler the current instance count instead of just disregarding the request.